src/neuroagent/agents/base_agent.py

```diff
@@ -93,7 +93,7 @@ async def arun(self, *args: Any, **kwargs: Any) -> AgentOutput:
         """Arun method of the service."""
 
     @abstractmethod
-    async def astream(self, *args: Any, **kwargs: Any) -> AsyncIterator[str]:
+    def astream(self, *args: Any, **kwargs: Any) -> AsyncIterator[str]:
```

Comment:
Interesting. Is this what helped you get ride of the `#type: ignore`?

I did not do any digging myself on it:)